### PR TITLE
Docker cluster enhancements, better provisioning error reporting

### PR DESCRIPTION
**Pull request by sethrosetter**
- [x] Ran `flake8`
- [x] Ran `run_repo_tests.sh`

#### Changes proposed in this pull request:

- Do not silently create docker networks with the same name
- Better error reporting when provisioning Couchbase Server

### CODE REVIEW

File: docker/cluster.py [hunk 1]
```diff
@@ -54,8 +54,20 @@ def create_cluster(network_name, number_of_nodes, public_key_path, dev, pull):
...
-    network = docker_client.networks.create(network_name)
+    log_info('Checking if network ({}) exists ...'.format(network_name))
+    conflicting_networks = [net for net in docker_client.networks.list() if net.name == 'cbl']
+
+    # If there is a network defined with the name of one you are trying to create
+    # Dev mode: reuse the existing network
+    # Strict mode: raise an exception
+    if len(conflicting_networks) > 0:
+        if not dev:
+            raise DockerError('ERROR!! Network already defined')
```
Comment (raghusarangapani) [inline docker/cluster.py]: Can we also provide an override to go ahead and delete the network and create a new one?

File: keywords/couchbaseserver.py [hunk 2]
```diff
@@ -196,15 +196,18 @@ def wait_for_ready_state(self):
...
 
-            if time.time() - start > keywords.constants.CLIENT_REQUEST_TIMEOUT:
-                raise Exception("Verify Docs Present: TIMEOUT")
+            elapsed = time.time()
+            if elapsed - start > keywords.constants.CLIENT_REQUEST_TIMEOUT:
+                raise Exception("Timeout: Server not in ready state! {}s".format(elapsed))
 
             # Verfy the server is in a "healthy", not "warmup" state
             try:
+                log_info("Going to: {}".format(self.url))
```
Comment (raghusarangapani) [inline keywords/couchbaseserver.py]: Is this a debug log?

File: keywords/couchbaseserver.py [hunk 3]
```diff
@@ -196,15 +196,18 @@ def wait_for_ready_state(self):
...
                 resp = self._session.get("{}/pools/nodes".format(self.url))
                 log_r(resp)
-            except ConnectionError:
+            except ConnectionError as e:
                 # If bringing a server online, there may be some connnection issues. Continue and try again.
+                log_info(e)
```
Comment (raghusarangapani) [inline keywords/couchbaseserver.py]: Is this a debug log?

File: docker/cluster.py (hunk 1 above)
Comment (sethrosetter) [inline docker/cluster.py]: `destroy_cluster` will do this.

**Pull request merged**